qiskit/quantum_info/operators/symplectic/pauli.py

```diff
@@ -146,6 +146,8 @@ class initialization (``Pauli('-iXYZ')``). A ``Pauli`` object can be
     # Set the max Pauli string size before truncation
     __truncate__ = 50
 
+    _VALID_LABEL_PATTERN = re.compile(r"^[+-]?1?[ij]?[IXYZ]+$")
```

Comment:
Could you add a comment on what pattern is matched? Regex is not always that straightforward to read for everyone 😉

Reply:
I dunno it's pretty verbose to put a plain text explanation on the regex. It'd be something like: "from the beginning of the string either 0 or 1 instances of either `+` or `-`, followed by a 0 or 1 instances  of `1` followed by 0 or 1 instance of `i` or `j`, followed by 1 or more of `I` `X` `Y` or `Z` to the end of the string."

That being said is `jIXY` an expected valid label because that would match this regex? (I would have expected only `1jIXY` to be valid)

Reply:
@Cryoris Yeah, I'm not good at parsing regex, so I'm ok with adding a comment here, on the other hand, I'd like to avoid verbose description as @mtreinish suggested.
How about this?
```
A valid Pauli label is a combination of a phase and a Pauli string. A phase string is a combination of at most three character, ['+', '-', ''], ['1', ''], and ['i', 'j', ''], e.g. '', '-1j'. A Pauli string is 1 or more characters of 'I' 'X' 'Y' or 'Z', e.g. 'Z', 'XIYY'.
```
(if it's ok) where should it be added? Docstring for `_VALID_LABEL_PATTERN` or Pauli class?

@mtreinish `jIXY` or `+1XY` is a valid label. I think it'd be better to keep the original specification for backward compat. I'm also satisfied with the current loose spec (some of them look a bit strange but at least the meaning is clear, so I think we have no reason to be strict here).

Reply:
I've added the docstring to `__init__`